R/measure.R

```diff
@@ -45,6 +45,7 @@ with_profiling <- function(profiling_on, expr) {
 }
 
 with_gc_info <- function(expr) {
+  force(expr)
```

Comment:
This is necessary because `bench::with_gcinfo()` uses `sink()`, which does strange things with unevaluated promises—it swallows the error and keeps going, even though we have code to prevent exactly that.